governator-core/src/main/java/com/netflix/governator/event/ApplicationEventModule.java

```diff
@@ -0,0 +1,106 @@
+package com.netflix.governator.event;
+
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.inject.AbstractModule;
+import com.google.inject.TypeLiteral;
+import com.google.inject.matcher.Matchers;
+import com.google.inject.spi.InjectionListener;
+import com.google.inject.spi.ProvisionListener;
+import com.google.inject.spi.TypeEncounter;
+import com.google.inject.spi.TypeListener;
+import com.netflix.governator.event.guava.GuavaApplicationEventModule;
+
+/**
+ * Adds support for passing {@link ApplicationEvent}s. Default (Guava-based) implementation
+ * can be found in {@link GuavaApplicationEventModule}
+ * 
+ * See {@link EventListener} and {@link ApplicationEventDispatcher} for usage. 
+ */
+public class ApplicationEventModule extends AbstractModule {
+    
+    private final ApplicationEventSubscribingTypeListener subscribingTypeListener; 
+    private final ApplicationEventDispatcher dispatcher;
+    
+    public ApplicationEventModule(ApplicationEventDispatcher dispatcher) {
+        this.dispatcher = dispatcher;
+        subscribingTypeListener = new ApplicationEventSubscribingTypeListener(dispatcher);
+    }
+    
+    private static class ApplicationEventSubscribingTypeListener implements TypeListener {
+
+        private static final Logger LOG = LoggerFactory.getLogger(ApplicationEventModule.class);
```

Comment:
Is this intended to be ApplicationEventModule.class or ApplicationEventSubscribingTypeListener.class?